ambari-server/src/main/resources/common-services/AMBARI_INFRA/0.1.0/package/scripts/params.py

```diff
@@ -106,15 +116,7 @@ def get_name_from_principal(principal):
   infra_solr_log = format("{infra_solr_log_dir}/solr-install.log")
   solr_env_content = config['configurations']['infra-solr-env']['content']
 
-zookeeper_port = default('/configurations/zoo.cfg/clientPort', None)
-# get comma separated list of zookeeper hosts from clusterHostInfo
-index = 0
-zookeeper_quorum = ""
-for host in config['clusterHostInfo']['zookeeper_hosts']:
-  zookeeper_quorum += host + ":" + str(zookeeper_port)
-  index += 1
-  if index < len(config['clusterHostInfo']['zookeeper_hosts']):
-    zookeeper_quorum += ","
+  zk_quorum = format(config['configurations']['infra-solr-env']['infra_solr_zookeeper_quorum'])
```

Comment:
Since `infra_solr_zookeeper_quorum` is not added on Ambari upgrade, shouldn't this use `default(..., zookeeper_quorum)` instead?

Reply:
in the infra_solr_zookeeper_quorum definition I added the on-ambari-upgarde add=true stuff.
Also I have to say i did not test the upgrade, but I expecting that field should be added during upgrade.